cg/meta/workflow/mutant/quality_controller/quality_controller.py

```diff
@@ -0,0 +1,265 @@
+            samples_quality_results
+        )
+
+        write_report(
+            case_qc_report_path=case_qc_report_path,
+            samples_quality_results=samples_quality_results,
+            case_quality_result=case_quality_result,
+        )
+
+        log_results(
+            case_quality_result=case_quality_result,
+            samples_quality_results=samples_quality_results,
+            report_file_path=case_qc_report_path,
+        )
+
+        summary: str = get_summary(
+            case_quality_result=case_quality_result,
+            samples_quality_results=samples_quality_results,
+        )
+
+        return QualityResult(
+            case_quality_result=case_quality_result,
+            samples_quality_results=samples_quality_results,
+            summary=summary,
+        )
+
+    def _get_samples_quality_results(
+        self, quality_metrics: QualityMetrics
+    ) -> SamplesQualityResults:
+        samples_quality_results: list[SampleQualityResults] = []
+        for sample in quality_metrics.pool.samples:
+            sample_results: SampleResults = quality_metrics.results[sample.internal_id]
+            sample_quality_results: SampleQualityResults = (
+                self._get_sample_quality_result_for_sample(
+                    sample=sample, sample_results=sample_results
+                )
+            )
+            samples_quality_results.append(sample_quality_results)
+
+        internal_negative_control_sample: Sample = quality_metrics.pool.internal_negative_control
+        internal_negative_control_quality_metrics: SampleQualityResults = (
+            self._get_sample_quality_result_for_internal_negative_control_sample(
+                sample=internal_negative_control_sample
+            )
+        )
+
+        external_negative_control_sample: Sample = quality_metrics.pool.external_negative_control
+        external_negative_control_sample_results: SampleResults = quality_metrics.results[
+            external_negative_control_sample.internal_id
+        ]
+        external_negative_control_quality_metrics: SampleQualityResults = (
+            self._get_sample_quality_result_for_external_negative_control_sample(
+                sample=external_negative_control_sample,
+                sample_results=external_negative_control_sample_results,
+            )
+        )
+
+        return SamplesQualityResults(
+            samples=samples_quality_results,
+            internal_negative_control=internal_negative_control_quality_metrics,
+            external_negative_control=external_negative_control_quality_metrics,
+        )
+
+    @staticmethod
+    def _get_sample_quality_result_for_sample(
+        sample: Sample, sample_results: SampleResults
+    ) -> SampleQualityResults:
+        sample_passes_reads_threshold: bool = sample_has_valid_total_reads(sample=sample)
+        sample_passes_qc: bool = sample_passes_reads_threshold and sample_results.qc_pass
```

Comment:
Similar to @ivadym's comment, it might be nice to call this `does_sample_pass_qc` or something